Index: plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js b/plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js
--- a/plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js	(revision aee1902ef2253f1dde9d8eaa1320be982d83459e)
+++ b/plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js	(revision 6a7d23ebde41ed3f2c7b31166c6b0cd642076dd0)
@@ -567,11 +567,11 @@
     var REBX = Math.max(0.5,round(Math.min(REBG60,REBG),2));
     var Hypo_ratio = 1;
 
-     if (currentTIRLow > 10 || AIMI_BreakFastLight || DeltaPctD < (-1.5) ){
+     if (currentTIRLow > 10 || AIMI_BreakFastLight || circadian_smb > (-3) ){
      var hypo_target = 100 * Math.max(1,circadian_sensitivity);
      enlog += "target_bg from "+target_bg+" to "+hypo_target+" because currentTIRLow > 5 : "+currentTIRLow+"\n";
 
-     target_bg = hypo_target+circadian_smb;
+     target_bg = circadian_smb > 5 ? 144 : hypo_target+circadian_smb;
      Hypo_ratio = 0.7;
      enlog += "Hypo_ratio : "+Hypo_ratio+"\n";
      C2 = (target_bg * 1.618)-(glucose_status.delta * 1.618);
@@ -594,7 +594,7 @@
      sensitivityRatio = round(sensitivityRatio,2);
      enlog +="Sensitivity ratio set to "+sensitivityRatio+" based on temp target of "+target_bg+";\n";
      sens = target_bg > min_bg * 1.10 ? sens * 1.618 : sens;
-     basal = profile.current_basal * sensitivityRatio;
+     basal = circadian_smb > 5 ? profile.current_basal / 2 : profile.current_basal * sensitivityRatio;
      basal = round_basal(basal, profile);
      if (basal !== profile_current_basal) {
          enlog +="Adjusting basal from "+profile_current_basal+" to "+basal+";\n";
