Index: plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js b/plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js
--- a/plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js	(revision 300d1d89364a109eb0c541cb1dc35231e39823ed)
+++ b/plugins/aps/src/main/assets/AIMI/b30mssvaimi/determine-basal.js	(revision 8cf36355f99a1b47a345c546b2a7d7c4913b8888)
@@ -1708,16 +1708,18 @@
              insulinReq = ((1 + Math.sqrt(delta)) / 4);
              var microBolus = circadian_smb > (-10) ? Math.min(AIMI_UAM_CAP,insulinReq*smb_ratio) :
              Math.min(AIMI_UAM_CAP,insulinReq);
-             }else if (iTimeActivation && !AIMI_BreakFastLight && !profile.temptargetSet && delta > 0 && aimismb === true && sens_predType == "UAM+" && UAMpredBG >= 150){
-             insulinReq = ((1 + Math.sqrt(delta)) / 2);
-                 if (circadian_smb > (-3)){
-                 var microBolus = Math.min(AIMI_UAM_CAP,insulinReq*smb_ratio);
-                 }else if (circadian_smb < (-5)){
-                 var microBolus = Math.min(AIMI_UAM_CAP,insulinReq*2);
-                 }else{
-                 var microBolus = Math.min(AIMI_UAM_CAP,insulinReq);
-                 }
-             }else if (delta > 0){
+             }else if (iTimeActivation && !AIMI_BreakFastLight && !profile.temptargetSet && delta > 0 && aimismb === true && UAMpredBG >= 150){
+               insulinReq = ((1 + Math.sqrt(delta)) / 2);
+                   if (circadian_smb > (-3)){
+                     var microBolus = Math.min(AIMI_UAM_CAP,insulinReq*smb_ratio);
+                     }else if (circadian_smb < (-5)){
+                     var microBolus = Math.min(AIMI_UAM_CAP,insulinReq*2);
+                     }else if(circadian_smb > -1 && bg > 200 ){
+                     var microBolus = Math.min(AIMI_UAM_CAP,round((Math.min(minPredBG,eventualBG) - target_bg) / future_sens, 2));
+                     }else{
+                     var microBolus = Math.min(AIMI_UAM_CAP,insulinReq);
+                     }
+               }else if (delta > 0){
              var microBolus = Math.min(insulinReq*smb_ratio, maxBolusTT);
              }
              }
